src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/TypeName.cs

```diff
@@ -392,5 +411,99 @@ public int GetArrayRank()
 #else
         ImmutableArray<TypeName> GetGenericArguments() => _genericArguments;
 #endif
+
+#if SYSTEM_REFLECTION_METADATA
+        /// <summary>
+        /// Creates a new <see cref="TypeName" /> object that represents current simple name with provided assembly name.
+        /// </summary>
+        /// <param name="assemblyName">Assembly name.</param>
+        /// <returns>Created simple name.</returns>
+        /// <exception cref="InvalidOperationException">The current type name is not simple.</exception>
+        public TypeName MakeSimpleTypeName(AssemblyNameInfo? assemblyName)
+        {
+            if (!IsSimple)
+            {
+                TypeNameParserHelpers.ThrowInvalidOperation_NotSimpleName(FullName);
+            }
+
+            TypeName? declaringType = IsNested
+                ? DeclaringType.MakeSimpleTypeName(assemblyName)
+                : null;
+
+            return new TypeName(fullName: FullName,
+                assemblyName: assemblyName,
+                elementOrGenericType: null,
+                declaringType: declaringType,
+                genericTypeArguments: ImmutableArray<TypeName>.Empty);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="TypeName" /> object representing a one-dimensional array
```

Comment:
Methods always return (well, except ones that [DoesNotReturn]...).  I think the verb you want here is "Creates", like in the previous method.

This applies to the remainder of the Make- methods.